vscode/src/commands/scm/source-control.ts

```diff
@@ -0,0 +1,248 @@
+                this.statusUpdate()
+            })
+
+            const { model, contextWindow } = this.modelProvider
+            const prompt = await buildPrompt(
+                contextWindow,
+                getSimplePreamble(model, 1, commitPrompts.intro),
+                await getContext(repository, commitTemplate).catch(() => [])
+            ).catch(error => {
+                sourceControlInputbox.value = `${error}`
+                throw new Error()
+            })
+
+            const stream = this.chatClient.chat(
+                prompt,
+                { model, maxTokensToSample: contextWindow.output },
+                abortController?.signal
+            )
+
+            // Function to update the input box with the latest text
+            const updateInputBox = (text: string, hasStopped = false) => {
+                sourceControlInputbox.value = text
+                hasStopped && this.statusUpdate()
+            }
+
+            await streaming(stream, abortController, updateInputBox, progress)
+        } catch (error) {
+            this.statusUpdate()
+            progress.report({ message: 'Error' })
+            if (error instanceof Error && error.message) {
+                sourceControlInputbox.value = initialInputBoxValue // Revert to initial value on error
+                vscode.window.showInformationMessage(`Generate commit message failed: ${error.message}`)
+            }
+        }
+    }
+
+    /**
+     * Updates the commit generation state and sets the corresponding context status.
+     * If an `abortController` is provided, it is used to abort the current commit generation.
+     *
+     * @param abortController - An optional `AbortController` instance to use for aborting the current commit generation.
+     */
+    private statusUpdate(abortController?: AbortController): void {
+        const isGenerating = abortController !== undefined
+        const contextID = 'cody.isGeneratingCommit'
+        vscode.commands.executeCommand('setContext', contextID, isGenerating)
+
+        this.abortController?.abort()
+        this.abortController = abortController
+    }
+
+    public dispose(): void {
+        for (const disposable of [...this.disposables]) {
+            disposable.dispose()
+        }
+        this.disposables = []
+    }
+}
+
+async function buildPrompt(
+    contextWindow: ModelContextWindow,
+    preamble: Message[],
+    context: ContextItem[]
+): Promise<Message[]> {
+    if (!context.length) {
+        throw new Error('Failed to get git output.')
+    }
+
+    const transcript: ChatMessage[] = [{ speaker: 'human', text: commitPrompts.message }]
+    const promptBuilder = new PromptBuilder(contextWindow)
+    promptBuilder.tryAddToPrefix(preamble)
+    promptBuilder.tryAddMessages(transcript.reverse())
+
+    const { ignored } = await promptBuilder.tryAddContext('user', context)
+    if (ignored.length) {
+        throw new Error('Cody cannot generate commit for ignored files.')
```

Comment:
```suggestion
        vscode.window.showWarningMessage('The commit message will not include changes made in ignored files.')
```

@RXminuS @umpox if we don''t want to block users from generating the commit when ignored files are detected

Reply:
Yeah, let's do this!

Reply:
updated by Rik